src/models/tag/stream.rs

```diff
@@ -22,6 +20,47 @@ pub enum TagStreamReach {
     Followers,
     Following,
     Friends,
+    MostFollowed,
+    Influencers,
+}
+
+impl TagStreamReach {
+    pub fn to_graph_subquery(&self) -> String {
+        let query = match self {
+            TagStreamReach::Followers => "MATCH (user:User)<-[:FOLLOWS]-(reach:User)",
+            TagStreamReach::Following => "MATCH (user:User)-[:FOLLOWS]->(reach:User)",
+            TagStreamReach::Friends => {
+                "MATCH (user:User)-[:FOLLOWS]->(reach:User), (user)<-[:FOLLOWS]-(reach)"
+            }
+            TagStreamReach::MostFollowed => "",
+            TagStreamReach::Influencers => "",
+        };
+        String::from(query)
+    }
+}
+
+#[derive(Deserialize, Debug, ToSchema, Clone)]
+pub enum TaggedType {
+    Post,
+    User,
+}
+
+pub struct HotTagsInput {
+    pub from: i64,
+    pub to: i64,
+    pub skip: usize,
+    pub limit: usize,
+    pub taggers_limit: usize,
+    pub tagged_type: Option<TaggedType>,
+}
```

Comment:
A mix of existing types (see `/src/types/pagination`) with custom fields needed for HotTags. If the idea behind this struct is to keep the function number of input params under control, let's aim to reuse the existing Pagination struct.

Reply:
The idea of `Pagination` is to be universally used across the `service` models and routes all the way down to the RedisOps trait and avoid as much "prop drilling" as possible. I think if we use it as well in routes that only have `skip` and `limit` like `user_following_handler()` for example.